src/store.rs

```diff
@@ -441,6 +444,12 @@ impl<F: LurkField> IntoHashComponents<F> for ScalarPtr<F> {
 #[derive(Debug, Clone, PartialEq, Eq)]
 pub struct ScalarContPtr<F: LurkField>(F, F);
 
+impl<F: LurkField> std::fmt::Display for ScalarContPtr<F> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "({:?}, {:?})", self.0, self.1)
```

Comment:
Here you’re using the debug implementation of both field and tag. If you rebase, you’ll see the first element of the ` ScalarContPtr` is a tag, which you can instead call display on. For the second element, ` hex_digits` is a better display.